core/src/main/java/org/elasticsearch/common/breaker/CircuitBreakingException.java

```diff
@@ -67,6 +68,11 @@ public long getByteLimit() {
     }
 
     @Override
+    public RestStatus status() {
+        return RestStatus.INTERNAL_SERVER_ERROR;
```

Comment:
I think this is justified as we want to enforce 500 no matter what. Otherwise we can run into the fragile base class problem. 

If we really want to remove the override, we have at least `NettyHttpRequestSizeLimitIT` as a safety guard which will break if the status code changes.